src/main/java/org/apache/commons/validator/routines/ISINValidator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.validator.routines;
+
+import java.io.Serializable;
+import java.util.Locale;
+import org.apache.commons.validator.routines.checkdigit.LuhnCheckDigit;
+
+/**
+ * International Securities Identification Number (ISIN) uniquely identifies a
+ * bonds, commercial paper, stocks or warrant.
+ */
+public class ISINValidator implements Serializable {
```

Comment:
There are several base classes in the routines package, which may be of help when implementing this. Have you looked at them? I don't know the validation logic of ISIN, but I could imagine that it has something to do with CheckDigits? Have a look at ISBNValidator and IBANValidator for examples.